storage/engine/rocksdb/db.cc

```diff
@@ -360,6 +384,54 @@ class DBCompactionFilterFactory : public rocksdb::CompactionFilterFactory {
   int64_t min_txn_ts_;
 };
 
+class DBComparator : public rocksdb::Comparator {
+ public:
+  DBComparator() {
+  }
+
+  virtual const char* Name() const override {
+    return "cockroach_comparator";
+  }
+
+  virtual int Compare(const rocksdb::Slice &a, const rocksdb::Slice &b) const override {
+    rocksdb::Slice key_a, key_b;
+    rocksdb::Slice ts_a, ts_b;
+    if (!SplitKey(a, &key_a, &ts_a) ||
+        !SplitKey(b, &key_b, &ts_b)) {
+      assert(false);
+      return a.compare(b);
```

Comment:
Yes, we do have asserts enabled by default. No, it never executes (unless we have some sort of corruption). I'll take the `assert(false)` out. It was leftover from some debugging I was doing.